### PR TITLE
Fix share via link Folder with no content

### LINKED ISSUE #11893: Do not show "Download" link is folder is empty
As seen in this screen shot, there is a "download" link in the top right corner. Upon clicking it, a user is able to download an empty directory. This is confusing and unhelpful. You should probably consider hiding it if the directory is empty

screen shot 2014-10-31 at 5 58 39 pm

### PR DESCRIPTION
**Pull request by davitol**
Fixes https://github.com/owncloud/core/issues/11893

Before the fix:

screen shot 2017-02-27 at 10 09 39

screen shot 2017-02-27 at 10 09 46

After the fix:

screen shot 2017-02-27 at 14 11 02

Pull request based in  https://github.com/owncloud/core/pull/24641 hints

### CODE REVIEW

File: apps/files_sharing/js/public.js [hunk 1]
```diff
@@ -222,9 +222,18 @@ OCA.Sharing.PublicApp = {
 			};
 
 			this.fileList.updateEmptyContent = function() {
+				var list = this;
 				this.$el.find('#emptycontent .uploadmessage').text(
 					t('files_sharing', 'You can upload into this folder')
 				);
+
+				$('#download').click(function(event) {
```
Comment (PVince81) [inline apps/files_sharing/js/public.js]: No, leave the event handler outside at line 248.

Doing it this way will re-re-re-re-register the event again every time this function is called, so clicking on the button will run the handler many many times.

**Comment by PVince81** [comment]
Can't test download button event because of a regression, fixed in this PR: https://github.com/owncloud/core/pull/27263

File: apps/files_sharing/js/public.js [hunk 2]
```diff
@@ -222,6 +222,7 @@ OCA.Sharing.PublicApp = {
 			};
 
 			this.fileList.updateEmptyContent = function() {
+				var list = this;
```
Comment (PVince81) [inline apps/files_sharing/js/public.js]: not needed any more.

Also it seems you removed too much, the code that switches the CSS of the button is gone too

**Comment by PVince81** [comment]
@davitol let me know when this is tested and ready for review, thanks

**Comment by PVince81** [comment]
any update ?

**Comment by davitol** [comment]
Close due to lack of time right now. Hopefully I'll reopen it when i'll found the way to make all the different scenarios I met work properly

**Pull request closed without merging**